aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2153,11 +2211,30 @@ impl AptosVM {
             &resolver,
             /*override_is_delayed_field_optimization_capable=*/ Some(false),
         );
+
         let log_context = AdapterLogSchema::new(state_view.id(), 0);
-        let mut gas_meter = match vm.make_standard_gas_meter(max_gas_amount.into(), &log_context) {
-            Ok(gas_meter) => gas_meter,
-            Err(e) => return ViewFunctionOutput::new(Err(anyhow::Error::msg(format!("{}", e))), 0),
+
+        let vm_gas_params = match get_or_vm_startup_failure(&vm.gas_params, &log_context) {
+            Ok(gas_params) => gas_params.vm.clone(),
+            Err(err) => {
+                return ViewFunctionOutput::new(Err(anyhow::Error::msg(format!("{}", err))), 0)
+            },
         };
+        let storage_gas_params =
+            match get_or_vm_startup_failure(&vm.storage_gas_params, &log_context) {
+                Ok(gas_params) => gas_params.clone(),
+                Err(err) => {
+                    return ViewFunctionOutput::new(Err(anyhow::Error::msg(format!("{}", err))), 0)
+                },
+            };
+
+        let mut gas_meter = make_prod_gas_meter(
+            vm.gas_feature_version,
+            vm_gas_params,
+            storage_gas_params,
+            false,
```

Comment:
nit: add /*is_governance_proposal = */ or similar before false? Some IDEs do it automatically it but for code review or things like emacs it is always better to have a comment saying what the constant represents